airflow-core/src/airflow/api_fastapi/core_api/routes/ui/grid.py

```diff
@@ -191,6 +191,8 @@ def grid_data(
                 task_node_map_exclude and ti.task_id not in task_node_map_exclude
             ) or ti.state == TaskInstanceState.REMOVED:
                 continue
+            if ti.task_id not in task_node_map:
+                continue
```

Comment:
Why do we need to add this?

Reply:
Initially, I tested it by deleting the dag_version record, which caused the API to fail. Now, I believe that this action is not necessary since it is a soft delete.